src/components/Game.js

```diff
@@ -0,0 +1,262 @@
+                      opacity: moves || games ? 0.75 : 0
+                    }}
+                    position={{
+                      x: x * config.SCALE + config.X_OFFSET,
+                      y: y * config.SCALE + config.Y_OFFSET,
+                      z: 0
+                    }}
+                    sound={{
+                      on: 'click',
+                      src: '#toggleSound',
+                      poolSize: 3,
+                      volume: 4
+                    }}
+                    events={{
+                      click: this.handleClick,
+                      mouseenter: this.playHoverSound
+                    }}
+                    event-set__1="_event: mouseenter; material.opacity: 0.99"
+                    event-set__2="_event: mouseleave; material.opacity: 0.75"
+                    animation__rotate="property: rotation; dur: 1000;
+                      easing: linear; loop: false; to: 360 0 0"
+                    animation__opacity="property: material.opacity; dur: 1500;
+                      easing: easeInSine; loop: false; from: 0; to: 0.75"
+                  />
+                )
+              )}
+
+              <Moves val={moves.toString()} />
+              <Timer time={timeElapsed} />
+            </Entity>
+          : <Entity
+              class="clickable"
+              geometry={{ primitive: 'plane', width: 3, height: 2 }}
+              material={{ color: 'white', opacity: 0 }}
+              position={{ x: 2, y: 2, z: -2 }}
+              text={{
+                value: 'START',
+                align: 'center',
+                anchor: 'align',
+                baseline: 'center',
+                width: 15,
+                letterSpacing: 5
+              }}
+              events={{
+                click: this.startGame
+              }}
+            />}
+
+        {!ready
+          ? <Entity
+              primitive="a-plane"
+              height="2"
+              width="8"
+              src="#logo"
+              material={{
+                color: 'white',
+                opacity: 0.95
+              }}
+              position={{
+                x: 2,
+                y: 5,
+                z: 0
+              }}
+            />
+          : <Entity />}
+
+        {!ready
+          ? <Entity>
+              {lights.map((row, y) =>
+                <Entity
+                  primitive="a-plane"
+                  height={randWidth}
+                  width={randWidth}
+                  src={`#${textures[y][1]}`}
+                  material={{
+                    color: 'white',
+                    opacity: Math.random()
+                  }}
+                  position={{
+                    x: -5 + Math.floor(Math.random() * 15),
+                    y: -2.5 + Math.floor(Math.random() * 15),
+                    z: Math.floor(Math.random() * 4)
+                  }}
+                />
+              )}
+            </Entity>
+          : <Entity />}
+
+        {hasWon
```

Comment:
These ternary operators all over the place are horrendous. For some reason, when I break them out into smaller, external components. The app breaks. This could be a flaw in the aframe-react bindings, but I need to investigate more.